src/main/java/org/openrewrite/java/spring/framework/MigrateResponseStatusExceptionGetStatusCodeMethod.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.spring.framework;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeTree;
+import org.openrewrite.java.tree.TypeUtils;
+
+public class MigrateResponseStatusExceptionGetStatusCodeMethod extends Recipe {
+    private static final String HTTP_STATUS_CLASS = "org.springframework.http.HttpStatus";
+    private static final String HTTP_STATUS_CODE_CLASS = "org.springframework.http.HttpStatusCode";
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate `ResponseStatusException#getStatus()` to `getStatusCode()`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrate Spring Framework 5.3's `ResponseStatusException` method `getStatus()` to Spring Framework 6's `getStatusCode()`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new CustomVisitor();
+    }
+
+    static class CustomVisitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            if (method.getMethodType() != null && method.getMethodType().getDeclaringType().getFullyQualifiedName().equals("org.springframework.web.server.ResponseStatusException") &&
+                    method.getSimpleName().equals("getStatus")) {
+                return JavaTemplate.builder("#{any()}.getStatusCode()")
+                        .javaParser(JavaParser.fromJavaVersion().classpathFromResources(executionContext, "spring-core-6", "spring-beans-6", "spring-web-6"))
+                        .build()
+                        .apply(getCursor(), method.getCoordinates().replace(), method.getSelect());
```

Comment:
The `type` of the `fieldType` of `method.getSelect` still references `HttpStatus` however I'm not sure if that's blocking the removal of the import